src/modules/cmdpal/Microsoft.CmdPal.UI/MainWindow.xaml.cs

```diff
@@ -525,8 +532,14 @@ private LRESULT HotKeyPrc(
         return PInvoke.CallWindowProc(_originalWndProc, hwnd, uMsg, wParam, lParam);
     }
 
-    private void AddNotificationIcon()
+    private void AddTrayIcon()
     {
+        var settings = App.Current.Services.GetService<SettingsModel>()!;
+        if (!settings.ShowSystemTrayIcon)
+        {
+            return;
+        }
```

Comment:
Do we need this tho other way around in RemoveTrayIcon too? Why only in AddTrayIcon?

Reply:
This is needed in `AddTrayIcon` because the method is also called on `WM_WINDOWPOSCHANGING` and `WM_TASKBAR_RESTART` messages.

Reply:
I changed the code to avoid reading settings twice.
Thanks for the feedback!